### PR TITLE
Support TLS/SSL over REST in the Java client

### LINKED ISSUE #16836: Provide TLS/SSL support for REST API calls in the Java client
### Description

Add support for TLS/SSL for the http client in the Zeebe Java client.

For the gRPC (TLS/SSL) configuration, the Zeebe Java client uses a `.pem` file which is loaded into the Netty client. The Apache HTTP Client doesn't support `.pem` files. To ensure a simple UX, with a single config property that doesn't break backwards compatibility, we should programmatically load the provided `.pem` file into a  Java KeyStore and pass it onto the Apache HTTP client.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Enable secure communication between the Zeebe Java client
and the Zeebe Gateway over REST by supporting TLS/SSL
configuration of the embedded HTTP client in the Zeebe
Java client.

## Related issues

closes #16836 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by npepinpe — APPROVED** [review]
:rocket:

File: zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java [hunk 1]
```diff
@@ -83,8 +103,16 @@ private HttpAsyncClientBuilder defaultClientBuilder() {
                 ", ",
                 ContentType.APPLICATION_JSON.getMimeType(),
                 ContentType.APPLICATION_PROBLEM_JSON.getMimeType()));
+
+    final HttpClientHostnameVerifier hostnameVerifier =
```
[the remark this answers is not in the document]
Comment (koevskinikola) [reply zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java:107]: I'll create a `tech debt` issue out of this as time is short before the `8.5` release.

**Comment by npepinpe** [comment]
Hmm, how do I find out which check failed on the queue?

**Comment by npepinpe** [comment]
Seems some unit and QA tests fail, possibly non flaky?

**Comment by npepinpe** [comment]
Nah, it's an atomix failing test, all good. Just bad luck.

**Comment by npepinpe** [comment]
Hm, but some QA tests for user tasks fail completely. We should look into that.

**Comment by npepinpe** [comment]
Seems like different checks failing, and today seems very prone to network issues...

**Comment by npepinpe** [comment]
So imo everything's fine.

**Pull request merged**